python/tests/__init__.py

```diff
@@ -20,11 +20,113 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 import base64
+import json
+import sys
+
+import msprime
+import numpy as np
 
 import tskit
 from . import tsutil
 from .simplify import *  # NOQA
 
+if msprime.__version__ <= "0.7.4":
```

Comment:
Is this robust to people running the development version of msprime? It'd be worth running the tests in a virtualenv to see. Quite a few devs are running the git version of msprime now.

Reply:
The tests fail with a dev version of msprime. I coded up https://github.com/benjeffery/msprime/tree/mutation_time_hack to fix this on the msprime end, but discovered the tskit tests already fail with a dev msprime see below, so maybe not worth the hack for now.
```
======================================================================
FAIL: test_draw_integer_breaks_ts (tests.test_drawing.TestDrawSvg)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/benj/projects/tskit/python/tests/test_drawing.py", line 1774, in test_draw_integer_breaks_ts
    self.assertNotEqual(svg.find(f">{b:.0f}<"), -1)
AssertionError: -1 == -1
-------------------- >> begin captured logging << --------------------
msprime.ancestry: DEBUG: Setting initial model {'name': 'hudson'}
msprime.ancestry: DEBUG: Running simulation until maximum: inf
--------------------- >> end captured logging << ---------------------

======================================================================
FAIL: test_integer_edges_all_sites (tests.test_topology.TestReduceTopology)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/benj/projects/tskit/python/tests/test_topology.py", line 6526, in test_integer_edges_all_sites
    ts = self.get_integer_edge_ts(5, 10)
  File "/home/benj/projects/tskit/python/tests/test_topology.py", line 6514, in get_integer_edge_ts
    self.assertEqual(int(edge.left), edge.left)
AssertionError: 6 != 6.016224464401603
-------------------- >> begin captured logging << --------------------
msprime.ancestry: DEBUG: Setting initial model {'name': 'hudson'}
msprime.ancestry: DEBUG: Running simulation until maximum: inf
--------------------- >> end captured logging << ---------------------

======================================================================
FAIL: test_integer_edges_one_site (tests.test_topology.TestReduceTopology)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/benj/projects/tskit/python/tests/test_topology.py", line 6519, in test_integer_edges_one_site
    ts = self.get_integer_edge_ts(5, 10)
  File "/home/benj/projects/tskit/python/tests/test_topology.py", line 6514, in get_integer_edge_ts
    self.assertEqual(int(edge.left), edge.left)
AssertionError: 6 != 6.016224464401603
-------------------- >> begin captured logging << --------------------
msprime.ancestry: DEBUG: Setting initial model {'name': 'hudson'}
msprime.ancestry: DEBUG: Running simulation until maximum: inf
--------------------- >> end captured logging << ---------------------
```

Reply:
Didn't see this comment sorry - GitHub is annoying! Once you comment on a PR it keeps updating on every tiny change that's made, so I mute it. But it appears this mutes the responses too.

Reply:
These tests are pretty trivial, it's just using some deprecated msprime APIs around discrete sites. Nothing to get  worried about.